addons/mail/static/src/messaging/entity/follower/follower.js

```diff
@@ -0,0 +1,207 @@
+odoo.define('mail.messaging.entity.Follower', function (require) {
+'use strict';
+
+const {
+    fields: {
+        many2many,
+        many2one,
+    },
+    registerNewEntity,
+} = require('mail.messaging.entity.core');
+
+function FollowerFactory({ Entity }) {
+
+    class Follower extends Entity {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @param {mail.messaging.entity.FollowerSubtype} subtype
+         */
+        checkSubtype(subtype) {
+            if (!this.checkedSubtypes.includes(subtype)) {
+                this.link({ checkedSubtypes: subtype });
+            }
```

Comment:
About the condition, what does link do if the relation already exists?

Can we avoid calling this method when the condition is false?

The caller should probably just do the link directly.
"check" is also not an ideal name (neither for the method nor for the relation), leaking the UI element that leads to here. If we make it a toggle (that is visually not a checkbox) it loses its meaning.
Maybe "selected"? Or "subscribed" (to reuse the route name).

Reply:
I would go for selected because subscribed is not really the case if it's not saved so you could have a subtype "checked" to which the follower is not subscribed (yet)